File: illchess-game-service/illchess-game/illchess-game-adapter/src/main/java/pl/illchess/game/adapter/game/command/in/rest/dto/MovePieceRequest.java
package pl.illchess.game.adapter.game.command.in.rest.dto;

import pl.illchess.game.application.game.command.in.MovePieceUseCase;

import java.util.UUID;

public record MovePieceRequest(
    UUID gameId,
    String startSquare,
    String targetSquare,
    String pawnPromotedToPieceType,
    String username
) {
    public MovePieceUseCase.MovePieceCmd toCmd() {
        return new MovePieceUseCase.MovePieceCmd(
            gameId,
            startSquare,
            targetSquare,
            pawnPromotedToPieceType,
            username
        );
    }
}

File: illchess-frontend/src/app/shared/model/game/MovePieceRequest.ts
import { Piece } from "./PieceInfo"

export interface MovePieceRequest {
    gameId: string
    startSquare: string
    targetSquare: string
    pawnPromotedToPieceType?: Piece.QUEEN | Piece.KNIGHT | Piece.ROOK | Piece.BISHOP,
    username: string
}